### PR TITLE
feat(image): drop canner image

### PR DESCRIPTION
**Pull request by neighborhood999**
Signed-off-by: Jie Peng <im@jiepeng.me>

**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

### CODE REVIEW

File: packages/client/src/admin/Images/ImageForm.tsx [hunk 1]
```diff
@@ -0,0 +1,765 @@
...
+            </label>
+            {form.getFieldDecorator('global', {
+              valuePropName: 'checked',
+              initialValue: globalStatus,
+            })(
+              <Switch
+                data-testid='global'
+                checkedChildren='Yes'
+                unCheckedChildren='No'
+                onChange={value => setGlobalStatus(value)}
```
Comment (aar0nTw) [inline packages/client/src/admin/Images/ImageForm.tsx:544]: why use state in boolean value?

Comment (neighborhood999) [reply packages/client/src/admin/Images/ImageForm.tsx]: Removed it.

File: packages/client/src/admin/Images/ImageAdd.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+          'pullSecret', // just only display on the custom image
+          'useImagePullSecret',
+        ]),
+        imageSpec: {
+          baseImage: restData.imageSpec.baseImage,
+          pullSecret: restData.imageSpec.pullSecret,
+          packages: {
+            apt: restData.apt.split('\n'),
+            conda: restData.conda.split('\n'),
+            pip: restData.pip.split('\n'),
```
Comment (ctiml) [inline packages/client/src/admin/Images/ImageAdd.tsx:55]: `packages.(apt|conda|pip)` should omit empty

current result:
```yaml
  packages:
    apt:
    - curl
    conda:
    - ""
    pip:
    - ""
```

expected:
```yaml
  packages:
    apt:
    - curl
```

**Comment by ctiml** [comment]
Comments for UI/UX

1. When editing image, the page will do reload every 10 secs. Could we by any chance removing the "reload" icon and the fading effect?
截圖 2021-09-03 14 43 30

2. UI consistency between "Images" and "Group images".
    1. Label
        - Images in Admin portal (this PR)
        截圖 2021-09-03 15 00 26
        - Group images in User portal
        截圖 2021-09-03 15 00 13
       
    2. Button position and style in detail page
        - Images (this PR)
        截圖 2021-09-03 15 08 11
        截圖 2021-09-03 15 14 17

        - Group images
        截圖 2021-09-03 15 09 06
        截圖 2021-09-03 15 14 05

**Comment by ctiml** [comment]
Rebuild button is not working

File: packages/client/src/admin/Images/ImageAdd.tsx (hunk 2 above)
Comment (neighborhood999) [reply packages/client/src/admin/Images/ImageAdd.tsx]: Updated, when one of the fields is empty, will send `null` instead of an empty string.

**Comment by neighborhood999** [comment]
> Comments for UI/UX
> 
>     1. When editing image, the page will do reload every 10 secs. Could we by any chance removing the "reload" icon and the fading effect?
>        截圖 2021-09-03 14 43 30
> 
>     2. UI consistency between "Images" and "Group images".
>        
>        1. Label
>           
>           * Images in Admin portal (this PR)
>             截圖 2021-09-03 15 00 26
>           * Group images in User portal
>             截圖 2021-09-03 15 00 13
>        2. Button position and style in detail page
>           
>           * Images (this PR)
>             截圖 2021-09-03 15 08 11
>             截圖 2021-09-03 15 14 17
>           * Group images
>             截圖 2021-09-03 15 09 06
>             截圖 2021-09-03 15 14 05

1. I removed the loading spinner.

File: packages/client/src/admin/Images/ImageAdd.tsx (hunk 2 above)
Comment (neighborhood999) [reply packages/client/src/admin/Images/ImageAdd.tsx]: Updated.

**Review by ctiml — APPROVED** [review]

**Review by aar0nTw — APPROVED** [review]
LGTM

**Pull request merged**